cilium/l7policy.cc

```diff
@@ -176,13 +177,28 @@ Http::FilterHeadersStatus AccessFilter::decodeHeaders(
 
 Http::FilterHeadersStatus AccessFilter::encodeHeaders(
     Http::ResponseHeaderMap& headers, bool) {
-  log_entry_.UpdateFromResponse(headers, config_->time_source_);
-  auto logType = ::cilium::EntryType::Response;
-  if (!allowed_ && headers.Status()->value() == "403") {
-    logType = ::cilium::EntryType::Denied;
-    config_->stats_.access_denied_.inc();
+  // Accepted & forwaded requests are logged by the upstream callback. Requests can remain unlogged
+  // if they are not accepted or any other error happens and upstream callback is never called.
+  // Logging the (locally generated) response is not enough as we no longer log request headers with
+  // responses.
+  if (!allowed_) {
+    // Request was not yet logged. Log it so that the headers get logged.
+    // Default logging local errors as "forwarded".
+    // The response log will contain the locally generated HTTP error code.
+    auto logType = ::cilium::EntryType::Request;
+
+    if (headers.Status()->value() == "403") {
+      // Log as a denied request.
+      logType = ::cilium::EntryType::Denied;
+      config_->stats_.access_denied_.inc();
+    }
+    config_->Log(log_entry_, logType);
   }
-  config_->Log(log_entry_, logType);
+
+  // Log the response
+  log_entry_.UpdateFromResponse(headers, config_->time_source_);
+  config_->Log(log_entry_, ::cilium::EntryType::Response);
```

Comment:
We now have two logs in the case of a not-allowed  request here. Should this be the case?

Reply:
Ah NVM. Actually I just realized that was the point. I didn't notice they were logging different values. We always logged response, now we also appropriately log the request.